packages/devtools_app/lib/src/screens/network/network_request_inspector_views.dart

```diff
@@ -135,23 +135,34 @@ class HttpRequestView extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final theme = Theme.of(context);
-    final requestHeaders = data.requestHeaders;
-    final requestContentType = requestHeaders?['content-type'] ?? '';
-    Widget child;
-    if (requestContentType.contains('json')) {
-      child = JsonViewer(encodedJson: data.requestBody!);
-    } else {
-      child = Text(
-        data.requestBody!,
-        style: theme.fixedFontStyle,
-      );
-    }
-    return Padding(
-      padding: const EdgeInsets.all(denseSpacing),
-      child: SingleChildScrollView(
-        child: child,
-      ),
+    return ValueListenableBuilder(
+      valueListenable: data.updateCount,
+      builder: (context, __, ___) {
+        final theme = Theme.of(context);
+        final requestHeaders = data.requestHeaders;
+        final requestContentType = requestHeaders?['content-type'] ?? '';
+        final isLoading = data.isFetchingFullData;
+        if (isLoading) {
+          return CenteredCircularProgressIndicator(
```

Comment:
Does this not have a const constructor?

Reply:
it won't allow it because mediumProgressSize isn't const, it is a getter :/